testify/test_discovery.py

```diff
@@ -31,18 +30,48 @@ class DiscoveryError(TestifyError):
     pass
 
 
-def gather_test_paths(testing_dir):
-    """Given a directory path, yield up paths for all py files inside of it"""
-    for adir, subdirs, subfiles in os.walk(testing_dir):
-        # ignore .svn directories and other such hiddens
-        if adir.startswith('.'):
+def to_module(path):
+    path = os.path.relpath(path)
+
+    if path.startswith('..'):
+        raise DiscoveryError('Test outside of current directory')
```

Comment:
This is starting to converge with some pytest code which I helped fix recently. Maybe you will find it useful:

https://bitbucket.org/logilab/pylint/src/c4231fff3e3429f76e07e9762d51f312c5322ede/lint.py?at=default#cl-1054

https://bitbucket.org/logilab/pylint/src/c4231fff3e3429f76e07e9762d51f312c5322ede/lint.py?at=default#cl-81

Here's the PR. Maybe it's more readable: https://bitbucket.org/logilab/pylint/commits/58c66aa083777059a2e6b46f6a0545a2f4977097